### PR TITLE
Moved Cluster.Sharding MNTK tests to a separate assembly

### PR DESCRIPTION
**Pull request by alexvaluyskiy**
- Moved Cluster.Sharding MNTK tests to a separate assembly
- MNTK tests should run in a specific order: Remote -> Cluster -> ClusterTools -> Sharding

### CODE REVIEW

**Comment by alexvaluyskiy** [comment]
@Aaronontheweb I've changed the order of MNTK tests, and the test from Cluster (LeaderElectionWithFailureDetectorPuppet) starts fail. So, It just random failing, doesn't related with Cluster.Tools or Cluster.

I've found this error
```
[ERROR][9/10/2016 8:28:13 AM][Thread 0021][akka://MultiNodeClusterSpec/system/log1-TestEventListener] Index (zero based) must be greater than or equal to zero and less than the size of the argument list.
Cause: System.FormatException: Index (zero based) must be greater than or equal to zero and less than the size of the argument list.
   at System.Text.StringBuilder.AppendFormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at System.String.FormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at System.String.Format(String format, Object[] args)
   at Akka.Event.DefaultLogMessageFormatter.Format(String format, Object[] args) in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka\Event\DefaultLogMessageFormatter.cs:line 23
   at Akka.Event.LogMessage.ToString() in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka\Event\LogMessage.cs:line 42
   at System.Text.StringBuilder.AppendFormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at System.String.FormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at System.String.Format(String format, Object[] args)
   at Akka.Event.LogEvent.ToString() in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka\Event\LogEvent.cs:line 65
   at Akka.Event.StandardOutLogger.PrintLogEvent(LogEvent logEvent) in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka\Event\StandardOutLogger.cs:line 132
   at Akka.Event.DefaultLogger.Print(LogEvent logEvent) in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka\Event\DefaultLogger.cs:line 39
   at Akka.TestKit.TestEventListener.Receive(Object message) in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka.TestKit\EventFilter\TestEventListener.cs:line 61
   at Akka.Actor.ActorBase.AroundReceive(Receive receive, Object message) in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka\Actor\ActorBase.cs:line 154
   at Akka.Actor.ActorCell.ReceiveMessage(Object message) in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka\Actor\ActorCell.DefaultMessages.cs:line 168
   at Akka.Actor.ActorCell.Invoke(Envelope envelope) in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka\Actor\ActorCell.DefaultMessages.cs:line 77
```

I've seen this error before... in Cluster.Tools tests

File: build.fsx [hunk 1]
```diff
@@ -308,13 +308,12 @@ Target "PrintMultiNodeTests" <| fun _ ->
...
 
 Target "MultiNodeTests" <| fun _ ->
-    let testSearchPath =
-        let assemblyFilter = getBuildParamOrDefault "spec-assembly" String.Empty
-        sprintf "src/**/bin/Release/*%s*.Tests.MultiNode.dll" assemblyFilter
-
     mkdir testOutput
     let multiNodeTestPath = findToolInSubPath "Akka.MultiNodeTestRunner.exe" "bin/core/Akka.MultiNodeTestRunner*"
-    let multiNodeTestAssemblies = !! testSearchPath
+    let multiNodeTestAssemblies = !! "src/**/bin/Release/Akka.Remote.Tests.MultiNode.dll" ++
```
Comment (alexvaluyskiy) [inline build.fsx]: We should run MNTK tests in a specific order: Remote -> Cluster -> ClusterTools

Comment (Aaronontheweb) [inline build.fsx]: Need to add the sharding tests to this

Comment (Aaronontheweb) [inline build.fsx]: But otherwise I agree with the change

File: src/core/Akka.Remote.TestKit/Player.cs [hunk 2]
```diff
@@ -402,7 +402,7 @@ public void InitFSM()
                     {
                         if (@event.StateData.RunningOp == null)
                         {
-                            _log.Warning("did not expect {1}", @event.FsmEvent);
+                            _log.Warning("did not expect {0}", @event.FsmEvent);
```
Comment (Aaronontheweb) [inline src/core/Akka.Remote.TestKit/Player.cs]: might be the fix for the bug you reported

File: build.fsx (hunk 1 above)
Comment (alexvaluyskiy) [inline build.fsx]: Sharding tests are not work at the moment. We could enable them later

Comment (alexvaluyskiy) [inline build.fsx]: They just exist

**Pull request merged**